src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -2063,11 +2063,45 @@ string ToModeString(FileSystemInfo fileSystemInfo)
         /// <returns>Name if a file or directory, Name -> Target if symlink.</returns>
         public static string NameString(PSObject instance)
         {
-            return instance?.BaseObject is FileSystemInfo fileInfo
-                ? InternalSymbolicLinkLinkCodeMethods.IsReparsePointWithTarget(fileInfo)
-                    ? $"{fileInfo.Name} -> {InternalSymbolicLinkLinkCodeMethods.GetTarget(instance)}"
-                    : fileInfo.Name
-                : string.Empty;
+            string[] executableExtensions = { ".exe", ".bat", ".com", ".cmd" };
```

Comment:
This is incomplete, needs to use [`CommandDiscovery.PathExtensions`](https://github.com/PowerShell/PowerShell/blob/18717c529b90f69f4e3ab67ca238100b84064e71/src/System.Management.Automation/engine/CommandDiscovery.cs#L1396), or [`CommandDiscovery.PathExtensionsWithPs1Prepended`](https://github.com/PowerShell/PowerShell/blob/18717c529b90f69f4e3ab67ca238100b84064e71/src/System.Management.Automation/engine/CommandDiscovery.cs#L1377).

Reply:
Good catch, will fix